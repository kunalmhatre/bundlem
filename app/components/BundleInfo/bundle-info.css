.bundle-info {
  font-weight: 700;
  border-bottom: 1px solid;
}

.bundle-info-title {
  padding: 1rem 0 0.5rem 0;
  font-size: 1.5rem;
}

.bundle-info-description {
  padding: 0.5rem 0 0.5rem 0;
  opacity: 0.7;
}

.bundle-info-resources-count {
  padding: 0.5rem 0 1rem 0;
}

@media only screen and (max-width: 767px) {
  .bundle-info-title {
    font-size: 1.2rem;
  }
  .bundle-info-description {
    font-size: 0.8rem;
  }
  .bundle-info-resources-count {
    font-size: 0.8rem;
  }
}