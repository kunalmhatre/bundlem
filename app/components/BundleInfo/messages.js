/*
 * BundleInfo Messages
 *
 * This contains all the text for the BundleInfo component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.BundleInfo';

export default defineMessages({
  resources: {
    id: `${scope}.resources`,
    defaultMessage: 'Resources: ',
  },
});
