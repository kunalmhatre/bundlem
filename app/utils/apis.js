/**
 * REST APIs
 */

export const domain = 'https://bundlem.herokuapp.com';

export const verifyTokenAPI = `${domain}/verify`;
export const publishBundleAPI = `${domain}/create`;
export const getBundleAPI = `${domain}/bundle`;
