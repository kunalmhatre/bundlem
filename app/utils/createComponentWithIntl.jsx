/**
 * Renders component with react-intl support for snapshot testing
 * Source: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#helper-function-2
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

const createComponentWithIntl = (children, props = { locale: 'en' }) => renderer.create(
  <IntlProvider {...props}>
    {children}
  </IntlProvider>,
);

export default createComponentWithIntl;
