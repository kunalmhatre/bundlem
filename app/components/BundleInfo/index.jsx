import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import messages from './messages';
import './bundle-info.css';

const propTypes = {
  bundleTitle: PropTypes.string,
  resourcesCount: PropTypes.number,
  bundleDescription: PropTypes.string,
};

const defaultProps = {
  bundleTitle: 'Default Title',
  resourcesCount: 0,
  bundleDescription: null,
};

/**
 * BundleInfo
 * @param {string} bundleTitle - Title of the bundle
 * @param {string} bundleDescription - Description of the bundle
 * @param {number} resourcesCount - Total resources included in the bundle
 */
function BundleInfo({
  bundleTitle,
  bundleDescription,
  resourcesCount,
}) {
  return (
    <div className="bundle-info theme-blue">
      <div className="bundle-info-title">{bundleTitle}</div>
      {
        bundleDescription ? (
          <div className="bundle-info-description">{bundleDescription}</div>
        ) : null
      }
      <div className="bundle-info-resources-count">
        <FormattedMessage {...messages.resources} />
        {resourcesCount}
      </div>
    </div>
  );
}

BundleInfo.propTypes = propTypes;
BundleInfo.defaultProps = defaultProps;

export default BundleInfo;
