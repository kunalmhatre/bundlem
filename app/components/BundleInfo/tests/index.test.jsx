import React from 'react';

import createComponentWithIntl from '../../../utils/createComponentWithIntl';
import { mountWithIntl } from '../../../utils/reactIntlHelperFunction';
import BundleInfo from '../index';

describe('<BundleInfo />', () => {
  const bundleTitle = 'testTitle';
  const resourceCount = 1337;

  it('renders old snapshot', () => {
    expect(
      createComponentWithIntl(
        <BundleInfo
          bundleTitle={bundleTitle}
          resourcesCount={resourceCount}
        />,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  it('shows bundle description when provided with one', () => {
    const testDescription = 'testDescription';

    expect(
      mountWithIntl(
        <BundleInfo
          bundleTitle={bundleTitle}
          resourcesCount={resourceCount}
          bundleDescription={testDescription}
        />,
      ).find('div.bundle-info-description')
        .text(),
    ).toBe(testDescription);
  });
});
