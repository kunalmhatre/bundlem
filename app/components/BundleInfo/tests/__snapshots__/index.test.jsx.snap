// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BundleInfo /> renders old snapshot 1`] = `
<div
  className="bundle-info theme-blue"
>
  <div
    className="bundle-info-title"
  >
    testTitle
  </div>
  <div
    className="bundle-info-resources-count"
  >
    <span>
      Resources: 
    </span>
    1337
  </div>
</div>
`;
